File: functions/libgcrypt/variants/go/1_ctz_1.go
package main

func ctz(in uint64) int32 {
	if in == 0 {
		return 64
	}

	var r int32
	for (in & 1) == 0 {
		r++
		in >>= 1
	}

	return r
}


func main(){
    ctz(0)
}

File: functions/libgcrypt/variants/c/1_ctz_1.c
#include <stdint.h>

int ctz(uint64_t in)
{
 if (in == 0) return 64;
 int r = 0;
 for (; (in & 1) == 0; in >>= 1) r++;
 return r;
}


int main(){
    ctz(0);
}